scripts/increment_auto_off.py

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env python3
+# A program that ticks a counter down until deactivating log_persistence
+
+import logging
+import sys
+import json
+import time
+
+import requests
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+sh = logging.StreamHandler(sys.stdout)
+logger.addHandler(sh)
+
+loop = True
+
+while loop:
+  try:
+    response = requests.get('http://localhost:5001/settings/persist_logs', timeout=10)
+    state = response.json()
+
+    if state["persist_logs"]:
+      future_persist_state = state["auto_off_delay"] != 1
+      delay = state["auto_off_delay"] - 1
+      body = {
+        "persist_logs": future_persist_state,
+        "auto_off_delay": delay if future_persist_state else 14,  # If no longer persisting, set to default
+      }
+      json_data = json.dumps(body)
+      response = requests.post(
+        url='http://localhost:5001/settings/persist_logs',
+        headers={'Content-Type': 'application/json'},
+        data=json_data,
+        timeout=10,
+      )
+
+      if response.ok:
+        loop = False
+        if future_persist_state:
+          logging.info(f"Persist logs will be automatically turned off in {delay} day(s)")
+        else:
+          logging.info("Persist logs has been turned off automatically")
+    else:
+      loop = False
+  except Exception as exc:
+    logger.exception(f"increment_auto_off.py has failed with the following exception, trying again in 5 seconds: {exc}")
+  finally:
+    if loop:
+      time.sleep(5)  # Loop until there is a response.ok at the end
```

Comment:
in this case, an endless loop is a bad idea. there is no indication that the error in question will suddenly resolve itself by virtue of trying again, since we catch all exceptions instead of ones that might actually work after retrying, such as anything `requests` raises. you also missed an entire logical branch where this could "fall through" and use some stale value.

even if it _is_ something that trying again could resolve, if it's not noticed in many days or months you could end up running hundreds of these if not caught. they should have an expiration date, at a bare minimum. 

looping to hit a webserver can exacerbate the issue; if the web server is not able to respond in a timely fashion, you could overload it with requests. that's especially problematic because of the above - you could end up with 300 scripts looping and hitting the web server over and over, and everything breaks down. that's especially sketchy because the server we're Denial-of-Service-ing has multiple ways to fix this when it happens (the support tunnel and a firmware upgrade), though a user is more liable to just restart their box before either of those. still, it's extremely bad in larger web app setups - your own webapp can DDOS your own webserver, if you ship it to your 10000 daily users (or at least make it balloon in scale and become very costly.)

I would personally not wrap any of this in a loop and [use a retry decorator with much better options](https://tenacity.readthedocs.io/en/latest/), especially for the exponential backoff and jitter options.